### DISCUSSION TITLE
USR-W610 Support

### DISCUSSION THREAD

**Discussion by alienatedsec** [alienatedsec/solis-ha-modbus-cloud#21] [Q&A] [answered]
Hi

Firstly great article and it clarified a few things for me re HA connectivity . Would it be possible to achieve the same thing if I used a single (USR-W610 Serial to WiFi Ethernet Wireless Converter RS232 RS485 Serial Server Support WatchDog Modbus Gateway TCP UDP) to replace the waveshare devices 

In essence I want to still use the s3 dongle if possible and use the 610 to integrate the inverter with with HA . I am not quite clear how the 485 wiring would be between the inverter  - W610 - S3 dongle 

Regards 

Nigel

_Originally posted by @wymes in https://github.com/alienatedsec/solis-ha-modbus-cloud/issues/2#issuecomment-1618205481_

**Comment by alienatedsec** [comment]
> In essence I want to still use the s3 dongle if possible and use the 610 to integrate the inverter with HA

@wymes you cannot achieve that by using only one device - I checked the documentation and this model also supports `Modbus TCP <> Modbus RTU`, along with `TCP Server` and `TCP Client` modes - https://www.pusr.com/support/download/[User-Manual]-USR-W610-datasheet-V1.0.1.html - your current device needs to be wired to RS485(A) and RS485(B) of the inverter and set as TCP Server.

You will need one more device to act as TCP Client and wired to RS485(A) and RS485(B) of your `S3-WIFI-ST` datalogger, and have it properly configured.

I can design wiring, but you need to work on the configuration as I have no device to test. We can start with what you already have, which will serve HA. Then, we could test if it works with your Datalogger on its own. Finally, you will need to purchase another device to combine both solutions.

Let me know if you want to proceed, but I don't guarantee it will work 100%.

**Reply by wymes** [reply, marked as answer]
Hi

I managed to get the w610 working with the inverter over RS485 and ha using the solax-modbus integration . Works a treat 
In the end I had to set the unit to transparent  mode and it burst into life I can connect over wifi to it no issue after it has joined my network .

next step is to use a second device to replicate your deplloyment

Thanks 

Nigel

**Comment by wymes** [comment]
Many thanks 

 

I will now purchase the W610 unit and investigate further . Just an off the wall thought . Given the unit also has a RS232 port  do you thinks I could use one of the following  to eliminate the second device by relaying the 485  to the 232 port  ?

 

https://www.amazon.co.uk/Converter-Adapter-Position-Terminal-Communication/dp/B08T9C22N9/ref=sr_1_6?keywords=rs232+to+rs485+converter&qid=1688399884&sr=8-6

 

This was on the amazon site  which helps understand the unit architecture 

 



 

 

Will keep you posted 

 

Regards

 

 

Nigel

**Comment by alienatedsec** [comment]
> Given the unit also has a RS232 port do you thinks I could use one of the following to eliminate the second device by relaying the 485 to the 232 port ?

@wymes I don't think there is a work mode that will allow RS232 and RS485 to coexist. Still, I might be wrong and can't test it.
